File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/UniqueItemsPriorityQue.java
package Auxiliary_Modules;

public class UniqueItemsPriorityQue extends LGFIFO<String>{
    // a priority queue without repeating elements
    @Override
    public void add(String item) {
        if(!super.contains(item)){
        super.add(item);}
    }
    public String peak() {
        String temp = super.peak();
        if (temp == null){
            return "";
        }
        return temp;
    }
    public Boolean strContainsResponse(String str){
        boolean result = false;
        for (String tempStr: this.elements) {
            if (str.contains(tempStr)){
                result = true;
                break;
            }
        }
        return result;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/UniqueItemsPriorityQue.kt
package auxiliary_modules

open class UniqueItemsPriorityQue : LGFIFO<String>() {
    // a priority queue without repeating elements
    override fun add(item: String) {
        if (!super.contains(item)) {
            super.add(item)
        }
    }

    override fun peak(): String {
        return super.peak() ?: return ""
    }

    fun strContainsResponse(str: String): Boolean {
        var result = false
        for (tempStr in elements) {
            if (str.contains(tempStr)) {
                result = true
                break
            }
        }
        return result
    }
}